enumerables.rb

```diff
@@ -0,0 +1,141 @@
+module Enumerable
+  def my_each
+    index = 0
+    while index < size
+      if is_a? Array
+        yield self[index]
+      elsif is_a? Range
+        yield to_a[index]
+      end
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    (0...size).each do |i|
+      yield(self[i], i)
+    end
+    self
+  end
+
+  def my_select
+    values = []
+    my_each do |val|
+      values.push(val) if yield(val)
+    end
+    values
+  end
+
+  def my_all?
+    if block_given?
+      values = true
+      my_each { |val| values = false unless yield(val) }
+      values
+    else
+      empty?
+    end
+  end
+
+  def my_any?
+    if block_given?
+      values = false
+      my_each { |val| values = true if yield(val) }
+      values
+    else
+      true
+    end
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains at least one truthy value.

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_any?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_any?(/w/)).`

- Your method doesn't consider the possibility when a string is given as an argument `(some_enumerable.my_any?('cow')).`